core/src/main/java/org/elasticsearch/common/transport/InetSocketTransportAddress.java

```diff
@@ -76,6 +56,12 @@ public InetSocketTransportAddress(InetAddress address, int port) {
     }
 
     public InetSocketTransportAddress(InetSocketAddress address) {
+        if (address == null) {
+            throw new IllegalArgumentException("InetSocketAddress must not be null");
+        }
+        if (address.getAddress() == null) {
+            throw new IllegalArgumentException("Address must be resolved but wasn't - InetSocketAddress#getAddress() returned null");
+        }
```

Comment:
+1 for these checks. Another way to accomplish the second check is to use address.isUnresolved(), but this is also fine.